addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaBridgeHandler.java

```diff
@@ -0,0 +1,283 @@
+                        // The user might have renamed some of the inputs etc.
+                        updateAllZoneInformation();
+                    }
+                }
+                    break;
+                default:
+                    logger.error("Channel " + id
+                            + " not supported on the yamaha device directly! Try with the zone things instead.");
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void refreshFromState(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case YamahaReceiverBindingConstants.CHANNEL_POWER:
+                updateState(channelUID, state.power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            default:
+                logger.error("Channel refresh for " + channelUID.getId() + " not implemented!");
+        }
+    }
+
+    /**
+     * Sets up a refresh timer (using the scheduler) with the CONFIG_REFRESH interval.
+     *
+     * @param initial_wait_time The delay before the first refresh. Maybe 0 to immediately
+     *            initiate a refresh.
+     */
+    private void setupRefreshTimer(int initial_wait_time) {
+        if (state == null) {
+            return;
+        }
+
+        Object interval_config_o = thing.getConfiguration().get(YamahaReceiverBindingConstants.CONFIG_REFRESH);
+        Integer interval_config;
+
+        if (interval_config_o == null) {
+            interval_config = refrehInterval;
+        } else {
+            interval_config = interval_config_o instanceof Integer ? (Integer) interval_config_o
+                    : ((BigDecimal) interval_config_o).intValue();
+        }
+
+        if (refreshTimer != null) {
+            refreshTimer.cancel(false);
+        }
+        refreshTimer = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                updateAllZoneInformation();
+            }
+        }, initial_wait_time, interval_config, TimeUnit.SECONDS);
+
+        refrehInterval = interval_config;
+    }
+
+    /**
+     * Periodically and initially called. Updates
+     */
+    void updateAllZoneInformation() {
+        try {
+            SystemControl basicDeviceInformation = new SystemControl();
+            basicDeviceInformation.fetchDeviceInformation(xml, state);
+            basicDeviceInformation.fetchPowerInformation(xml, state);
+
+            updateProperty(YamahaReceiverBindingConstants.PROPERTY_VERSION, state.version);
+            updateProperty(YamahaReceiverBindingConstants.PROPERTY_ASSIGNED_NAME, state.name);
+
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            state.invalidate();
+            return;
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
```

Comment:
It's unfortunately not that easy because some older firmware avrs send unexpected responses to some specific commands but all the rest of the functionality is still working. I don't want to put the device offline because of that (I actually implemented like this because a user had exactly this problem)